File: tests/conformance/fixtures/elixir/23-task-type.exs
use Sykli

pipeline do
  task "build" do
    run "go build ./..."
    task_type :build
  end

  task "test" do
    run "go test ./..."
    task_type :test
    after_ ["build"]
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()

File: tests/conformance/fixtures/rust/23-task-type.rs
use sykli::{Pipeline, TaskType};

fn main() {
    let mut p = Pipeline::new();

    p.task("build")
        .run("go build ./...")
        .task_type(TaskType::Build);
    p.task("test")
        .run("go test ./...")
        .task_type(TaskType::Test)
        .after(&["build"]);

    p.emit();
}
